### DISCUSSION TITLE
Toggle mic indicator or mic with indicator

### DISCUSSION THREAD

**Discussion by timurseleznev** [Sapd/HeadsetControl#312] [Q&A]
Hi, is it possible to switch the microphone indicator or microphone software? this is necessary in order to use hotkeys on the keyboard and understand whether the microphone is on now on or not.

**Comment by Sapd** [comment]
> microphone indicator 

I think I tested that one on a corsair void and in principle its possible (but might not be true for every headset). But you would have to find out the right packet for the specific device to incoperate into headsetcontrol.

But headsetcontrol itself is not able to switch it by itself, so basically one would need to make an additional script which watches the mute status, and then call headsetcontrol. (but thats probably easy, the hard part is finding the packet)

**Reply by timurseleznev** [reply]
Hi, thanks for your answer, I launched wireshark and found out that volume control and microphone mute do not interact with the dongle in any way. Only chatmix management packages are caught.

**Reply by Sapd** [reply]
So with packet I meant the packet for toggling the light (I assume you mean the indicator light).

The microphone mute and volume controls are transmitted always via the dongle, but always via the Audio interface. That means to fetch that you would have to write a script which fetches it with for example a ALSA library on Linux.

**Reply by timurseleznev** [reply]
I'm not good at this, if you can tell me how to choose a command to switch this indicator, I'd appreciate it.

**Reply by Sapd** [reply]
Its really not easy and highly depends on the headset. 
There is probably a packet which does it. If you do a WireShark on the headset and then try to press mute multiple times, is there really no packet sent **to** the headset? Bc. one of them must be switching the light.
